### PR TITLE
Adds new CSV report for user data

### PR DESCRIPTION
**Pull request by TheDancingClown**
## 📝 A short description of the changes

* Adds Reports::User class to generate report data for different user types. The Admin/Assessor/Judge user types are separate tables and have different columns - where a column does not exist "N/A" is returned.
* Adds roles methods to Judge and Assessor classes to return a list of Award types.
* Adds Reports::Admin::AssessorJudgeAdminDataReport class to map csv headers and data.
* Adds Admin::UsersReportsController and route to handle download and adds link to index view.

## 🔗 Link to the relevant story (or stories)

* https://app.asana.com/0/1200504523179343/1206833235918007

## :shipit: Deployment implications

* None

## ✅ Checklist

- [x] Features that cannot go live are behind a feature flag/env var or specify deploy date and open PR as draft 
- [x] I have checked that commit messages make sense and explain the reasoning for each change
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have squashed any unnecessary or part-finished commits

## 🖼️ Screenshots (if appropriate - no PII/Prod data):

Screenshot 2024-03-20 at 16 27 32

### CODE REVIEW

File: app/models/assessor.rb [hunk 1]
```diff
@@ -149,6 +149,12 @@ def lead_roles
     end
   end
 
+  def roles
+    FormAnswer::POSSIBLE_AWARDS.select do |cat|
+      get_role(cat) == "lead" || "regular"
```
Comment (dreamfall) [inline app/models/assessor.rb:154]: this is not gonna work, and always gonna return all roles.

File: app/models/reports/user.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  def user_type
+    if @user.is_a?(Admin)
+      "Admin"
+    elsif @user.is_a?(Assessor)
+      "Assessor"
+    elsif @user.is_a?(Judge)
+      "Judge"
+    else
+      "User"
+    end
```
Comment (dreamfall) [inline app/models/reports/user.rb:30]: can do just `@user.class.name`

File: app/models/reports/user.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def awards_assigned
+    if @user.is_a?(Assessor) || @user.is_a?(Judge)
+      @user.roles.reject{|e| e=~ /promotion/}.join(", ")
```
Comment (dreamfall) [inline app/models/reports/user.rb:35]:
```suggestion
      @user.roles.reject{ |e| e=~ /promotion/ }.join(", ")
```

**Review by dreamfall — CHANGES_REQUESTED** [review]

File: app/models/assessor.rb (hunk 1 above)
Comment (TheDancingClown) [reply app/models/assessor.rb:154]: Yes, I want to return the award types for which the assessor is either a regular or lead assessor. I want to omit awards they have not been assigned to.

Comment (dreamfall) [reply app/models/assessor.rb:154]: @TheDancingClown i mean this condition is always true, does not matter what the role is or if there is one.

Comment (TheDancingClown) [reply app/models/assessor.rb]: Ah I see - good spot, thanks

**Review by dreamfall — APPROVED** [review]

**Pull request merged**